core/src/main/java/org/apache/accumulo/core/dataImpl/TabletIdImpl.java

```diff
@@ -36,7 +36,7 @@ public int compareTo(TabletId o) {
 
   @Override
   public Text getTableId() {
-    return new Text(ke.getTableId().getUtf8());
+    return new Text(ke.getTableId().canonical());
```

Comment:
Would be nice if we deprecated this method for a new one that returns the new TableId type.  I think its confusing to have two different types returned across our API.

Reply:
The more we can deprecate Text from our API, the better.

Reply:
I opened up a follow on issue. https://github.com/apache/accumulo/issues/953